oak_functions/sdk/test_utils/src/lib.rs

```diff
@@ -180,6 +189,75 @@ where
     }
 }
 
+/// Builds the crate identified by the given name as a Linux binary, and returns the path of the
+/// resulting binary.
+fn build_rust_crate_linux(crate_name: &str) -> anyhow::Result<String> {
+    duct::cmd!(
+        "cargo",
+        "build",
+        "--target=x86_64-unknown-linux-musl",
+        "--release",
+        format!("--package={crate_name}"),
+    )
+    .dir(env!("WORKSPACE_ROOT"))
+    .run()
+    .context(format!("could not compile {crate_name}"))?;
+    Ok(format!(
+        "{}target/x86_64-unknown-linux-musl/release/{crate_name}",
+        env!("WORKSPACE_ROOT")
+    ))
+}
+
+/// Builds the crate identified by the given name as a Wasm module, and returns the path of the
+/// resulting binary.
+pub fn build_rust_crate_wasm(crate_name: &str) -> anyhow::Result<String> {
+    duct::cmd!(
+        "cargo",
+        "build",
+        "--target=wasm32-unknown-unknown",
+        "--release",
+        format!("--package={crate_name}"),
+    )
+    .dir(env!("WORKSPACE_ROOT"))
+    .run()
+    .context(format!("could not compile {crate_name}"))?;
+    Ok(format!(
+        "{}target/wasm32-unknown-unknown/release/{crate_name}.wasm",
+        env!("WORKSPACE_ROOT"),
+    ))
+}
+
+pub fn create_and_start_oak_functions_server(
+    server_port: u16,
+    wasm_module_path: &str,
+    lookup_data_path: &str,
+) -> anyhow::Result<duct::ReaderHandle> {
+    let oak_functions_launcher_path = build_rust_crate_linux("oak_functions_launcher")?;
+    let oak_functions_linux_fd_bin_path = build_rust_crate_linux("oak_functions_linux_fd_bin")?;
+
+    let handle = duct::cmd!(
+        oak_functions_launcher_path,
+        format!("--wasm={wasm_module_path}"),
+        format!("--port={server_port}"),
+        format!("--lookup-data={lookup_data_path}"),
+        "native",
+        format!("--app-binary={oak_functions_linux_fd_bin_path}"),
+    )
+    .dir(env!("WORKSPACE_ROOT"))
+    .reader()
+    .context("could not run oak_functions_launcher")?;
+    Ok(handle)
+}
+
+pub fn kill_process(handle: duct::ReaderHandle) {
```

Comment:
Can we add some documentation on what we expect to be killed afterwards?

Reply:
Done.